docs/src/idiomatic/assignments.md

```diff
@@ -0,0 +1,129 @@
+# Idiom 1: Update state variables with assignments
+For each TLA+ variable `x` defined in a specification, Apalache requires that every possible  transition defines an assignment to `x'`.
+
+## Assignments and Assignment Candidates
+An _assignment candidate_ for `x` is an expression that has one of the following forms:
+- `x' = e`, 
+- `x' \in e`,
+- `UNCHANGED x`, or
+- `x' := e`
+
+While a transition may contain multiple assignment candidates for the same variable, not all of them are chosen as _assignments_ by Apalache. The sections below describe how the assignments are selected.
+
+### Minimality
+Assignments aren't spurious; each variable must have at least one assignment per transition operator, but no more than necessary to satisfy all of the additional constraints below.
+
+If all possible transitions fail to assign one or more variables, an exception of the following shape is thrown:
+```
+Assignment error: No assignments found for: x, z 
+```
+
+Such exceptions are usually the result of adding a `VARIABLE` without any accompanying TLA+ code relating to it. 
+The case where at least one transition, but not all of them, fails to assign a variable is shown below.
+
+### Syntax Order
+For the purpose of evaluating assignments, Apalache considers the left-to-right syntax order of conjuncts. 
+Therefore, as many assignments as possible are selected from the first (w.r.t. syntax order) conjunct, then from the second, and so on.
+
+Example:
+```tla
+Next == x' = 1 /\ x' = 2
+```
+In the above example, `x' = 1` would be chosen as an assignment to `x`, over `x' = 2`.
+
+### Assignment-before-use Paradigm
+If, in the syntax order defined above, an expression containing a primed variable `x'` precedes an assignment to `x`, the assignment finder throws an exception of the following shape:
+```
+Assignment error: test.tla:10:16-10:17: x' is used before it is assigned.
+```
+notifying the user of any variables used before assignment. In particular, right-hand-sides of assignment candidates are subject to this restriction as well. Consider:
+```tla
+A == x' > 0 /\ x' = 1
+B == y' = x' + 2 /\ x' = 1
+```
+In `A`, the expression `x' > 0` precedes any assignment to `x` and in `B`, while `y' = x' + 2` is an assignment candidate for `y`, it precedes any assignment to `x`, so both expressions are inadmissible (and would trigger exceptions). 
+
+Note that this only holds true if `A` (resp. `B`) is chosen as the transition operator. If `A` is called inside another transition operator, for example in `Next == x' = 1 /\ A`, no exception is thrown.
```

Comment:
"Transition operator" is really a non-standard terminology. I know what you are talking about, but a normal reader would be lost.